src/BlazorAdmin/Services/CachedCatalogItemServiceDecorator.cs

```diff
@@ -0,0 +1,114 @@
+﻿using Blazored.LocalStorage;
+using BlazorShared.Interfaces;
+using BlazorShared.Models;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace BlazorAdmin.Services
+{
+    public class CachedCatalogItemServiceDecorator : ICatalogItemService
+    {
+        private readonly ILocalStorageService _localStorageService;
+        private readonly CatalogItemService _catalogItemService;
+        private ILogger<CachedCatalogItemServiceDecorator> _logger;
+
+        public CachedCatalogItemServiceDecorator(ILocalStorageService localStorageService,
+            CatalogItemService catalogItemService,
+            ILogger<CachedCatalogItemServiceDecorator> logger)
+        {
+            _localStorageService = localStorageService;
+            _catalogItemService = catalogItemService;
+            _logger = logger;
+        }
+
+        public async Task<List<CatalogItem>> ListPaged(int pageSize)
+        {
+            string key = "items";
+            var cacheEntry = await _localStorageService.GetItemAsync<CacheEntry<List<CatalogItem>>>(key);
+            if (cacheEntry != null)
+            {
+                _logger.LogInformation("Loading items from local storage.");
+                if (cacheEntry.DateCreated.AddMinutes(1) > DateTime.UtcNow)
+                {
+                    return cacheEntry.Value;
+                }
+                else
+                {
+                    _logger.LogInformation("Cache expired; removing types from local storage.");
```

Comment:
items not types
maybe the log messages should use the key, like $"Loading {key} from local storage" ?

Reply:
Done